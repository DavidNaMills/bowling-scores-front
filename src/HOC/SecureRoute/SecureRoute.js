import React, { Component } from 'react';
import {Route, Redirect} from 'react-router-dom';

import {useSelector} from 'react-redux';


const SecureRoute = ({component: Component, ...rest}) =>{
    console.log('secure');
    const token = useSelector(state=>state.user.token);
    return (
        <Route {...rest} component={(props)=>(
            token 
            ? <Component {...props} />
            : <Redirect to='/' />
        )} />
    )
}

export default React.memo(SecureRoute);