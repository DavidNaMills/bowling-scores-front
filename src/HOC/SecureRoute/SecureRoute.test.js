import React from 'react';
import {mount, render} from 'enzyme';
import renderer from 'react-test-renderer';
import {Route, Redirect, BrowserRouter} from 'react-router-dom';
import SecureRoute from './SecureRoute';


jest.mock('react-redux');
import {useSelector} from 'react-redux';

const mockToken = '45fg6dbd3b4v56c';
const TestComp = () =>(<div><p>test comp</p></div>);

afterEach(()=>{
    jest.clearAllMocks();
});

describe('SecureRoute test suite', ()=>{
    it('exists', ()=>{
        useSelector.mockImplementation(()=>(mockToken));
        const wrapper = mount(<BrowserRouter><SecureRoute component={TestComp}/></BrowserRouter>);
        expect(wrapper.exists).toBeTruthy();
        expect(wrapper.find(Route).length).toBe(1);
    })
    
    it('matches snapshot', ()=>{
        useSelector.mockImplementation(()=>(mockToken))
        const comp = renderer.create(<BrowserRouter><SecureRoute component={TestComp}/></BrowserRouter>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    })
    
    it('renders the component if token is present in the store', ()=>{
        useSelector.mockImplementation(()=>(mockToken))
        const wrapper = mount(<BrowserRouter><SecureRoute component={TestComp}/></BrowserRouter>);
        expect(wrapper.find(TestComp).length).toBe(1);
    })
    
    it('renders redirect if token in not present', ()=>{
        useSelector.mockImplementation(()=>(null))
        const wrapper = mount(<BrowserRouter><SecureRoute component={TestComp}/></BrowserRouter>);
        expect(wrapper.find(Redirect).length).toBe(1);
    })
});