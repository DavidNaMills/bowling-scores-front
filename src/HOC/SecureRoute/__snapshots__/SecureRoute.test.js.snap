// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecureRoute test suite matches snapshot 1`] = `
<div>
  <p>
    test comp
  </p>
</div>
`;
